server/utils/checkYourAnswersUtils.ts

```diff
@@ -42,28 +51,34 @@ export const getTaskAnswersAsSummaryListItems = (task: string, application: Appl
       task,
       pageKey,
       questions,
+      checkYourAnswersPage,
     })
   })
 
   return items
 }
 
 export const addPageAnswersToItemsArray = (params: {
-  items: Array<SummaryListItem>
+  items: Array<SummaryListItem | QuestionAndAnswer>
   application: Application
   task: string
   pageKey: string
   questions: Record<string, unknown>
+  checkYourAnswersPage: boolean
 }) => {
-  const { items, application, task, pageKey, questions } = params
+  const { items, application, task, pageKey, questions, checkYourAnswersPage } = params
   const PageClass = getPage(task, pageKey)
 
   const page = new PageClass({}, application)
 
   if (hasResponseMethod(page)) {
     const response = page.response()
     Object.keys(response).forEach(question => {
-      items.push(summaryListItemForQuestion(application, task, pageKey, { question, answer: response[question] }))
+      if (checkYourAnswersPage) {
+        items.push(summaryListItemForQuestion(application, task, pageKey, { question, answer: response[question] }))
+      } else {
```

Comment:
see above, regarding the `checkYourAnswers` boolean argument - but I wonder if maintainers would find it clearer to see something like `else if 'document'` here?